src/Analyzer.Utilities.UnitTests/JsonPathResolverTests.cs

```diff
@@ -255,6 +235,174 @@ public void ResolveResourceType_JObjectWithResourcesNotArrayOfObjects_ReturnsEmp
+                        ""Child"": [ 0, 1, 2 ],
+                        ""Child2"": {
+                            ""LastProp"": true
+                        }
+                    },
+                }");
+
+            var resolver = new JsonPathResolver(jobject, jobject.Path);
+
+            // Resolve root and compare to confirm that resolver is using the passed JObject directly (not copied).
+            var rootResult = resolver.Resolve("").ToList();
+            Assert.AreEqual(1, rootResult.Count);
+            Assert.AreEqual<object>(jobject, rootResult[0].JToken, "Root token does not match original JObject - a copy may have been returned.");
+
+            // Setup multiple paths to resolve to test cache retrieval
+            string[] pathsToResolve = ["NoChildren", "OneChildLevel.Child", "twochildlevels.child2.lastprop"];
+
+            // Create a 2D array to hold resolved results for each path and each resolution attempt
+            List<IJsonPathResolver>[,] resolvedResults = new List<IJsonPathResolver>[2, pathsToResolve.Length];
+
+            // Resolve each path twice. Track each result in a 2D array and compare later to confirm the resolver is using the internal cache correctly.
+            for (int i = 0; i < 2; i++)
+            {
+                for (int j = 0; j < pathsToResolve.Length; j++)
+                {
+                    // Resolve the specific path and confirm the correct property was resolved and resolver returns correct path
+                    resolvedResults[i, j] = resolver.Resolve(pathsToResolve[j]).ToList();
+
+                    Assert.AreEqual(1, resolvedResults[i, j].Count);
+                    Assert.AreEqual(pathsToResolve[j], resolvedResults[i, j][0].JToken.Path, ignoreCase: true);
+                    Assert.AreEqual(pathsToResolve[j], resolvedResults[i, j][0].Path, ignoreCase: true);
+                }
+
+                // Remove all properties to confirm cached paths are used for repeat lookup.
+                jobject.RemoveAll();
+
+                // Create a new resolver to confirm paths are now unavailable
+                var resolverWithEmptyJObject = new JsonPathResolver(jobject, jobject.Path);
+
+                for (int j = 0; j < pathsToResolve.Length; j++)
+                {
+                    var emptyResults = resolverWithEmptyJObject.Resolve(pathsToResolve[j]).ToList();
+                    Assert.AreEqual(1, emptyResults.Count);
+                    Assert.IsNull(emptyResults[0].JToken);
+                }
+            }
+
+            // Validate that the results from the first and second resolution of each path are the same.
+            for (int i = 0; i < pathsToResolve.Length; i++)
+            {
+                Assert.AreEqual(1, resolvedResults[0, i].Count);
+                Assert.AreEqual(1, resolvedResults[1, i].Count);
+                Assert.AreEqual<object>(resolvedResults[0, i][0].JToken, resolvedResults[1, i][0].JToken);
+            }
+        }
+
```

Comment:
Are there cases where both lang v1 and v2 can be declared in the same file? Would it cause issues?
What about in the same repo with a mix of nested templates in both languages?